### PR TITLE
use browsersleuth instead of browsers.contralis.info

### PR DESCRIPTION
**Pull request by fippo**
less moving parts wrt #34

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -43,13 +43,13 @@ if [[ $BVER =~ $VERSION_REGEX ]] ; then
...
   # of the website changes then it won't work anymore. We should add safari to
-  # browsers.contralis.info instead
+  # browsers.contralis.info / browser-sleuth instead
   TARGET_URL=`curl https://developer.apple.com/safari/download/ | sed -nE 's/.*href="(.*\.dmg)">.*macOS 10.12.*/\1/p'`
   TARGET_VERSION=`curl https://developer.apple.com/safari/download/ | sed -nE 's/.*>([0-9]+)<\/p>.*$/\1/p'`
 else
-  TARGET_BROWSER=`curl -H 'Accept: text/csv' http://browsers.contralis.info/$PLATFORM/$BROWSER/$BVER`
-  TARGET_URL=`echo $TARGET_BROWSER | cut -d',' -f7`
-  TARGET_VERSION=`echo $TARGET_BROWSER | cut -d',' -f5`
+  TARGET_BROWSER=`node_modules/browser-sleuth/sleuth.sh $BROWSER $BVER $PLATFORM`
```
Comment (markandrus) [inline setup.sh]: This causes an issue with the way I use travis-multirunner. I do something like

```
    cd node_modules/travis-multirunner
    if [ "${TRAVIS_OS_NAME}" == 'linux' ]; then
      BROWSER=chrome ./setup.sh
      BROWSER=firefox ./setup.sh
      export CHROME_BIN=$(pwd)/browsers/bin/chrome-$BVER
      export FIREFOX_BIN=$(pwd)/browsers/bin/firefox-$BVER
    else
      BROWSER=safari ./setup.sh
    fi
```

which gives the following error:

```
++node_modules/browser-sleuth/sleuth.sh chrome beta linux
./setup.sh: line 50: node_modules/browser-sleuth/sleuth.sh: No such file or directory
+TARGET_BROWSER=
```

Comment (markandrus) [reply setup.sh]: I think you could replace the hardcoded path with something like

```bash
SLEUTH=`node -e 'console.log(require.resolve("browser-sleuth/sleuth.sh"))'`
TARGET_BROWSER=`$SLEUTH $BROWSER $BVER $PLATFORM`
```

and the usage I described would continue to work.

Comment (fippo) [reply setup.sh]: or alternatively write a little script inside travis-multirunner that simplifies the split even

Comment (fippo) [reply setup.sh]: went your route

**Comment by fippo** [comment]
thoughts on this? i'm trying to cut cut down CI job run time and at 10s total runtime an extra http request becomes a low hanging fruit :-)

**Comment by DamonOehlman** [comment]
I think it's worth exploring, however, it won't get you away from having HTTP requests as part of the build process as `browser-sleuth` queries the firefox and chrome endpoints to determine version anyway.

I think Nathan's original intention when implementing [his API](https://github.com/nathanoehlman/browser-version-api) was to provide a one endpoint to "rule them all".

IMO, it's 6 in one hand and half a dozen in the other (assuming the browser version api remains up).  I'll create an issue on Nathan's browser version api project for us to discuss uptime, and I think if we don't find ourselves with a good solution, then maybe dropping back to using browser sleuth directly might be the right way to go.

**Comment by fippo** [comment]
see newer approach ;-)

**Pull request closed without merging**